### PR TITLE
Implement a Dark mode for Sheikah

### LINKED ISSUE #1571: Implementing Dark mode

### LINKED ISSUE #1601: Improve visual feedback for which page is selected in transactions pagination

### PR DESCRIPTION
**Pull request by gabaldon**
This pr implements a component to change the theme to a dark mode. It creates variables related to each theme.

Captura de pantalla 2021-01-12 a las 15 40 16

close #1571 
close #1601

### CODE REVIEW

**Comment by aesedepece** [comment]
image

Highligh color in select boxes is the same as foreground text color, so it makes the text in the options invisible.

**Comment by aesedepece** [comment]
image

Loading screen is hard to read now.

**Comment by Tommytrg** [comment]
image
Disabled buttons look like enabled ones.

**Review by aesedepece — APPROVED** [review]
All my former comments have been successfully addressed. No further objections from my side.

File: src/components/RadonAggregateTallyScript.vue [hunk 1]
```diff
@@ -9,21 +9,28 @@
       <div class="operator-bottom">
         <div class="icon-container">
           <img
-            v-if="emptyScript"
-            class="row sheikah-icon"
+            v-if="emptyScript && darkMode"
```
Comment (Tommytrg) [inline src/components/RadonAggregateTallyScript.vue:12]: Could you move this to a component?

File: src/components/RadonAggregateTallyScript.vue [hunk 2]
```diff
@@ -9,21 +9,28 @@
...
+            v-if="!emptyScript && !darkMode"
             src="@/resources/svg/operator-arrow.svg"
           />
           <div
             v-if="emptyScript"
             class="add-operator-container"
             @click="addOperator"
           >
-            <img class="add-operator" src="@/resources/svg/add-operator.svg" />
+            <img v-if="darkMode" src="@/resources/svg/add-operator-dark.svg" />
```
Comment (Tommytrg) [inline src/components/RadonAggregateTallyScript.vue:32]: Could we move this to a component?

File: src/components/RadonOperator.vue [hunk 3]
```diff
@@ -8,10 +8,17 @@
       <el-tooltip
         v-show="showDelete && !hideDelete"
         content="Delete operator"
-        placement="right"
+        placement="left"
         effect="light"
       >
         <img
+          v-if="darkMode"
```
Comment (Tommytrg) [inline src/components/RadonOperator.vue:15]: Could we move this to a component?

File: src/components/RadonScript.vue [hunk 4]
```diff
@@ -9,12 +9,22 @@
       <div class="operator-bottom">
         <div class="icon-container">
           <img
-            v-if="emptyScript"
+            v-if="emptyScript && darkMode"
```
Comment (Tommytrg) [inline src/components/RadonScript.vue:12]: Could we move this to a component?

File: src/components/RadonScript.vue [hunk 5]
```diff
@@ -23,7 +33,18 @@
             class="add-operator-container"
             @click="addOperator"
           >
-            <img class="add-operator" src="@/resources/svg/add-operator.svg" />
+            <img
```
Comment (Tommytrg) [inline src/components/RadonScript.vue:36]: Could we move this to a component?

File: src/components/Select.vue [hunk 6]
```diff
@@ -37,16 +37,28 @@
             />
             <div class="arrow">
               <img
-                v-if="areOptionsVisible"
+                v-if="areOptionsVisible && darkMode"
```
Comment (Tommytrg) [inline src/components/Select.vue:40]: Could we move this to a component?

File: src/components/SettingsThemeSwitch.vue [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+</template>
+
+<script>
+import { mapState, mapMutations, mapActions } from 'vuex'
+
+export default {
+  name: 'NotificationsSettings',
+  data() {
+    return {
+      activeTheme: 'Dark',
```
Comment (Tommytrg) [inline src/components/SettingsThemeSwitch.vue:18]:
```suggestion
      themeNames: {
      dark: 'Dark',
      light: 'Light'
      }
```

File: src/components/SettingsThemeSwitch.vue [hunk 8]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <el-switch
+    v-model="toggle"
+    class="notification"
+    data-test="action-try"
+    :active-text="activeTheme"
+    :inactive-text="inactiveTheme"
```
Comment (Tommytrg) [inline src/components/SettingsThemeSwitch.vue:7]:
```suggestion
    :active-text="themeNames.dark"
    :inactive-text="themeNames.light"
```

File: src/components/SettingsThemeSwitch.vue [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    return {
+      activeTheme: 'Dark',
+      inactiveTheme: 'Light',
+    }
+  },
+  computed: {
+    ...mapState({
+      darkMode: state => state.wallet.darkMode,
+    }),
+    toggle: {
```
Comment (Tommytrg) [inline src/components/SettingsThemeSwitch.vue:26]:
```suggestion
    activeTheme: {
```

File: src/components/SettingsThemeSwitch.vue [hunk 10]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <el-switch
+    v-model="toggle"
```
Comment (Tommytrg) [inline src/components/SettingsThemeSwitch.vue:3]:
```suggestion
    v-model="activeTheme"
```

File: src/store/wallet.js [hunk 11]
```diff
@@ -60,6 +62,7 @@ export default {
     exportFileLink: '',
     checkTokenGenerationEventDate: new Date(GENESIS_EVENT_TIMESTAMP),
     mainnetReady: false,
+    darkMode: false,
```
Comment (Tommytrg) [inline src/store/wallet.js:65]: With all the changes you made, we could add easily any theme. To make the most this, I would use a variable called "theme" witch value could be "light" | "dark". With this approach we could update or add a theme easily

File: src/components/Alert.vue [hunk 12]
```diff
@@ -1,122 +0,0 @@
-<template>
```
Comment (Tommytrg) [inline src/components/Alert.vue:1]: Spite we need to delete this component, I think it's out the scope of this big PR

**Review by Tommytrg — CHANGES_REQUESTED** [review]
Awesome job, just some considerations:
- Use theme = 'dark' instead of darkTheme = true to be able to add a new theme in the future.
- Make components that handle the theme instead to avoid add unnecessary logic to parent components.
- Avoid make changes that are out of the scope of the PRs to facilitate reviews :cry:

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**